src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -978,6 +1007,40 @@ def has_args(call_node: ast.Call, required_arg_ids: Tuple[str, ...]) -> bool:
     return False
 
 
+def merge_entry_points(
+    *all_entry_points_with_descriptions_of_source: Tuple[str, Dict[str, Dict[str, str]]]
+) -> Dict[str, List[str]]:
+    """Merge all entry points, throwing ValueError if there are any conflicts."""
+    # keep source_name info for each entry_point until we have merged all
+    # sources, so we can provide better error messages in case of conflicts
+    merged = cast(
+        # this gives us a two level deep defaultdict with the inner values being of list type
+        Dict[str, Dict[str, List[Tuple[str, str]]]],
```

Comment:
Ah, this type hint helps a ton! Imo you can get rid of the comment, the type hint is self-documenting.

Reply:
Type hint wise, I think you want to use `typing.DefaultDict`.

Reply:
> Ah, this type hint helps a ton! Imo you can get rid of the comment, the type hint is self-documenting.
you sure? the type hint was there all along :D